sycl/test/basic_tests/parallel_for_range.cpp

```diff
@@ -0,0 +1,93 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+int main() {
+  auto asyncHandler = [](exception_list es) {
+    for (auto& e : es) {
+      std::rethrow_exception(e);
```

Comment:
I am curious about what happens if there are more than 1 exception here.
Is there any chance we can throw the second one?

Reply:
In the specific way the tests are written, there should only be at most one async exception being thrown. (Actually, the exceptions shouldn't even be async, but currently exceptions thrown during queue::submit are async exceptions instead of sync exceptions--that's for a different PR, though).

With the way the test is written, only the first exception can be thrown.

Reply:
Yes, this is confusing...